product-scenarios/1-manage-users-roles/1.1-user-registration/1.1.2-user-registration-by-admin/1.1.2.1-user-registration-through-apis/1.1.2.1.2-scim-2.0/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserWithValidationFailuresTestCase.java

```diff
@@ -31,52 +29,56 @@
+import org.wso2.identity.scenarios.commons.util.Constants;
 
 import static org.testng.Assert.assertEquals;
-import static org.testng.Assert.assertNotNull;
 import static org.testng.Assert.assertTrue;
 
+
 public class ProvisionUserWithValidationFailuresTestCase extends ScenarioTestBase {
 
     private CloseableHttpClient client;
     private String PASSWORD = "ab";
+    private String scimUsersEndpoint;
+    private final String SEPERATOR = "/";
+    JSONArray schemasArray;
+
+    HttpResponse response;
 
     @BeforeClass(alwaysRun = true)
     public void testInit() throws Exception {
 
         setKeyStoreProperties();
         client = HttpClients.createDefault();
+        super.init();
     }
 
-    @Test(description = "1.1.1.1.3")
+    @Test(description = "1.1.2.1.2.3")
     public void testInvalidSCIMUserCreate() throws Exception {
 
-        String scimEndpoint = getDeploymentProperties().getProperty(IS_HTTPS_URL) + SCIMConstants.SCIM2_USERS_ENDPOINT;
-        HttpPost request = new HttpPost(scimEndpoint);
-        request.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
-        request.addHeader(HttpHeaders.CONTENT_TYPE, SCIMConstants.CONTENT_TYPE_APPLICATION_JSON);
-
         JSONObject rootObject = new JSONObject();
         JSONArray schemas = new JSONArray();
+
         rootObject.put(SCIMConstants.SCHEMAS_ATTRIBUTE, schemas);
         JSONObject names = new JSONObject();
         names.put(SCIMConstants.GIVEN_NAME_ATTRIBUTE, SCIMConstants.GIVEN_NAME_CLAIM_VALUE);
         rootObject.put(SCIMConstants.NAME_ATTRIBUTE, names);
         rootObject.put(SCIMConstants.USER_NAME_ATTRIBUTE, SCIMConstants.USERNAME);
         rootObject.put(SCIMConstants.PASSWORD_ATTRIBUTE, PASSWORD);
 
-        StringEntity entity = new StringEntity(rootObject.toString());
-        request.setEntity(entity);
+        response = SCIMProvisioningUtil.provisionUserSCIM(backendURL, rootObject, Constants.SCIMEndpoints.SCIM2_ENDPOINT,
+                Constants.SCIMEndpoints.SCIM_ENDPOINT_USER, ADMIN_USERNAME, ADMIN_PASSWORD);
 
-        HttpResponse response = client.execute(request);
-        assertEquals(response.getStatusLine().getStatusCode(), 500,
+        assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_INTERNAL_SERVER_ERROR,
                 "Password validation failed at user creation hence server should have returned a bad request");
 
         Object responseObj = JSONValue.parse(EntityUtils.toString(response.getEntity()));
         EntityUtils.consume(response.getEntity());
-        JSONArray schemasArray = (JSONArray) ((JSONObject) responseObj).get("schemas");
-        assertNotNull(schemasArray);
-        assertEquals(schemasArray.get(0).toString(), SCIMConstants.ERROR_SCHEMA);
+        JSONArray schemasArray = new JSONArray();
+        schemasArray.add(responseObj);
+
+        assertEquals(((JSONObject) responseObj).get("schemas"), SCIMConstants.ERROR_SCHEMA);
```

Comment:
Shall we use the 3 parameter assertEquals method, so a message will print if not equal